kms/capi/capi.go

```diff
@@ -0,0 +1,754 @@
+//go:build windows
+// +build windows
+
+package capi
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"github.com/pkg/errors"
+	"go.step.sm/crypto/kms/apiv1"
+	"go.step.sm/crypto/kms/uri"
+	"go.step.sm/crypto/randutil"
+	"golang.org/x/crypto/cryptobyte"
+	"golang.org/x/crypto/cryptobyte/asn1"
+	"golang.org/x/sys/windows"
+	"io"
+	"math/big"
+	"reflect"
+	"strings"
+	"unsafe"
+)
+
+// Scheme is the scheme used in uris.
+const Scheme = "capi"
+
+const (
+	ProviderNameArg  = "provider"
+	ContainerNameArg = "key"
+	HashArg          = "sha1"
+	StoreLocationArg = "store-location" // 'machine', 'user', etc
+	StoreNameArg     = "store"          // 'MY', 'CA', 'ROOT', etc
+	KeyIDArg         = "key-id"
+	SerialNumberArg  = "serial"
+	IssuerNameArg    = "issuer"
+)
+
+var signatureAlgorithmMapping = map[apiv1.SignatureAlgorithm]string{
+	apiv1.UnspecifiedSignAlgorithm: "ECDSA_P256",
+	apiv1.SHA256WithRSA:            "RSA",
+	apiv1.SHA384WithRSA:            "RSA",
+	apiv1.SHA512WithRSA:            "RSA",
+	apiv1.ECDSAWithSHA256:          "ECDSA_P256",
+	apiv1.ECDSAWithSHA384:          "ECDSA_P384",
+	apiv1.ECDSAWithSHA512:          "ECDSA_P521",
```

Comment:
You have constants in ncrypt_windows.go for those strings. It would make sense to use them here too.